packages/appium/lib/main.js

```diff
@@ -172,13 +169,34 @@ async function main (args = null) {
     // otherwise parse from CLI
     args = parser.parse_args();
   }
+
+  const configResult = await readConfigFile(args.config);
+  // TBD: maybe just warn?
+  if (!_.isEmpty(configResult.errors)) {
+    throw new Error(`Errors in config file ${configResult.filepath}:\n ${configResult.reason ?? configResult.errors}`);
+  }
+
+  // merge config and apply defaults.
+  // the order of precendece is:
+  // 1. command line args
+  // 2. config file
+  // 3. defaults from config file
+  args = _.defaults(args, configResult.config?.server, getDefaultsFromSchema({prop: 'server'}));
+
   await logsinkInit(args);
 
+  if (configResult.config) {
+    const configFilepath = path.relative(process.cwd(), configResult.filepath);
```

Comment:
is `filepath` always relative?

Reply:
I'm not sure I understand this question.  any filepath is always relative to the current working directory.  I can change this to display as absolute, but as far as I recall this was just for output.  config files _tend_ to be close to the cwd, practically speaking, so I thought it appropriate to display this way.